packages/ember-data/lib/system/store.js

```diff
@@ -1683,6 +1683,10 @@ Store = Service.extend({
       return internalModels.map((internalModel) => internalModel.getRecord());
     }
 
+    if (data.data === null) {
+      return null;
+    }
+
     var internalModel = this._pushInternalModel(data.data || data);
```

Comment:
I believe this works because the the `return null;` never gets hit if `data.data` is undefined. I'd like to remove this `|| data` from this line if possible. It seems like we could do that if we pull one of the assertions out of `_pushInternalModel` and put it into this method.